egs/wsj/s5/steps/nnet3/lstm/train.sh

```diff
@@ -451,6 +453,12 @@ while [ $x -lt $num_iters ]; do
   ilr=$initial_effective_lrate; flr=$final_effective_lrate; np=$num_archives_processed; nt=$num_archives_to_process;
   this_learning_rate=$(perl -e "print (($x + 1 >= $num_iters ? $flr : $ilr*exp($np*log($flr/$ilr)/$nt))*$this_num_jobs);");
 
+  if [ "$adaptive_shrink" == "true" ]; then
+    this_shrink=$(perl -e "print $shrink + ($flr*$this_num_jobs/$this_learning_rate) * (1 - $shrink)")
```

Comment:
this_learning_rate is a function of the num_initial_jobs, so to make this_shrink independent of num_initial_jobs we actually need

this_shrink=$(perl -e "print 1.0 - (($this_learning_rate / ($num_jobs_initial * $ilr)) * (1 - $shrink));")